Code Exposure Initial Commit

sjaws opened the pull request:
Code Exposure initial commit

*Issue #, if available:*

*Description of changes:*


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

sjaws committed:
Code Exposure Initial Commit

Code Exposure initial commit

kihakuni commented on docs/Code Exposure.md:
```diff
@@ -0,0 +1,261 @@
+It is highly recommended to export logs to a security incident event management (SIEM) solution (such as Splunk, ELK stack, etc.) to aid in viewing and analyzing a variety of logs for a more complete attack timeline analysis.
+
+### CloudTrail: Public S3 Bucket
+By default, CloudTrail logs API calls that were made in the last 90 days, but not log requests made to objects. You can see bucket-level events on the CloudTrail console. However, you can't view data events (Amazon S3 object-level calls) there—you must parse or query CloudTrail logs for them. 
```
"there—you must parse or query CloudTrail logs for them. "

Recommend changing this to "by default--you must enable object level logging before those events appear in CloudTrail."

sjaws replied:
Added recommended change.

kihakuni commented on docs/Code Exposure.md:
```diff
@@ -0,0 +1,261 @@
+1. Review CloudTrail logs for the eventnames `GetPublicAccessBlock` and `DeletePublicAccessBlock`
+
+### CloudTrail: Public S3 Object
+You can also get CloudTrail logs for object-level Amazon S3 actions. To do this, enable data events for your S3 bucket or all buckets in your account. When an object-level action occurs in your account, CloudTrail evaluates your trail settings. If the event matches the object that you specified in a trail, the event is logged. 
```
Recommend adding the process to enable events OR linking "enable data events" to the process in our documentation so it's clear how to do so

sjaws replied:
Added recommended link.

kihakuni commented on docs/Code Exposure.md:
```diff
@@ -0,0 +1,261 @@
+This playbook outlines the process to identify owners of code resources, determine how they were exposed, who may have accessed the exposed code, determine impact of exposure, required remediation actions, and determine root cause of code exposure.
+
+## Potential Indicators of Compromise
+- DLP Alerts
+- Known code publication or exposure
+- Suspicious CloudTrail logs
```
I see DLP and code publication mentioned here, but not elsewhere in the document? 

Did you want to focus on S3, or include these in procedures as well for a more complete document?

sjaws replied:
Added DLP and CloudTrail details.

kihakuni commented on docs/Code Exposure.md:
```diff
@@ -0,0 +1,261 @@
+### Incident Classification & Handling
+* **Tactics, techniques, and procedures**: Data Exfiltration, AWS Service Abuse
+* **Category**: Data Loss
+* **Resource**: CodeCommit, S3, EC2
```
I see CodeCommit mentioned here. Do we have incident response or review procedures for CodeCommit? 

Document focuses on S3, but I didn't see anything related to reviewing CodeCommit events beyond asset inventory

sjaws replied:
Added content for CodeCommit and EC2.

kihakuni commented on docs/Code Exposure.md:
```diff
@@ -0,0 +1,261 @@
+* **Category**: Data Loss
+* **Resource**: CodeCommit, S3, EC2
+* **Indicators**: Cyber Threat Intelligence, Third Party Notice, CloudWatch Metrics
+* **Log Sources**: DNS Logs, VPC Flow Logs, CloudTrail, CloudWatch
```
Is this accurate? Or are you only covering Cloudtrail and VPC Flow Logs in this document?

sjaws replied:
Added content for additional resources and log sources.

kihakuni commented on docs/Code Exposure.md:
```diff
@@ -0,0 +1,261 @@
+
+## Detection
+
+### Logging and S3 Bucket Checks
+- Ensure CloudTrail is enabled in all regions: `./prowler -c check21`
+- Ensure CloudTrail log file validation is enabled: `./prowler -c check22`
+- Ensure S3 bucket access logging is enabled on the CloudTrail S3 bucket: `./prowler -c check26`
+- Ensure there are no S3 buckets open to Everyone or Any AWS user: `./prowler -c extra73`
+- Identify the resources in your organization and accounts; such as Amazon S3 buckets or IAM roles; that are shared with an external entity: `./prowler -c extra769`
+- Find resources exposed to the internet: `./prowler -g group17`
```
Do we have similar things we can look for security checks against CodeCommit? 
For example, this git project? https://aws.amazon.com/blogs/compute/discovering-sensitive-data-in-aws-codecommit-with-aws-lambda-2/

sjaws replied:
While I am unaware of prowler checks for CodeCommit, I did add CodeCommit logging checks.

kihakuni commented on docs/Code Exposure.md:
```diff
@@ -0,0 +1,261 @@
+You can also review [Blocking public access to your Amazon S3 storage](https://aws.amazon.com/s3/features/block-public-access/) for additional details on blocking public S3 access across your account.
+
+## Eradication
+
```
Did you want to include processes/documents for EC2 and CodeCommit?

sjaws replied:
Added documentation for EC2 and CodeCommit.

kihakuni commented on docs/Code Exposure.md:
```diff
@@ -0,0 +1,261 @@
+Same procedures as those listed for Eradication
+
+## Preventative Actions
+
```
Any preventative actions for EC2 and CodeCommit?

sjaws replied:
Added documentation for EC2 and CodeCommit.

sjaws committed:
Implementing AWS feedback

Adding more details, links to AWS documentation and CodeCommit specific references.

sjaws commented:
Updated pull request.

sjaws committed:
Branding changes

code commit is CodeCommit

kihakuni committed:
Update Code Exposure.md

Spelling corrections, minor grammar changes
